meta/rpc.go

```diff
@@ -0,0 +1,445 @@
+package meta
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/influxdb/influxdb/meta/internal"
+)
+
+// Max size of a message before we treat the size as invalid
+const (
+	MaxMessageSize    = 1024 * 1024 * 1024
+	leaderDialTimeout = 10 * time.Second
+)
+
+// RPC handles request/response style messaging between cluster nodes
+type RPC struct {
+	Logger         *log.Logger
+	tracingEnabled bool
+
+	store interface {
+		cachedData() *Data
+		IsLeader() bool
+		Leader() string
+		Peers() []string
+		AddPeer(host string) error
+		CreateNode(host string) (*NodeInfo, error)
+		NodeByHost(host string) (*NodeInfo, error)
+		WaitForDataChanged() error
+	}
+}
```

Comment:
Ah right..  I think I intended for all this to be private at one point but some is public inadvertently.  I'll make fix it up and either make it all private or all public.